clang/test/SemaSYCL/allow-constexpr-recursion.cpp

```diff
@@ -84,7 +83,6 @@ void constexpr_recurse_test() {
 }
 
 void constexpr_recurse_test_err() {
-  // expected-error@+1{{SYCL kernel cannot call a recursive function}}
   int i = constexpr_recurse(1);
```

Comment:
Shouldn't this actually diagnose? @erichkeane can you please check.

Reply:
Yes, exactly.  SEe my comment above.

Reply:
> Yes, exactly. SEe my comment above.

I'm looking at what's going on with this function call. The call count is 0 so not detected as recursive. Also "visit call" comes here 2 times. not sure why there are 2 visits, haven't debugged that yet. On both visits 'count' is 0. On one of the visits ConstexprDepth is 0 and the other ConstexprDepth is 1.  Separately: so-- if a function is marked constexpr, yet the compiler is allowed to not evaluate the call until execution time?

to make my debugging easier, i have rewritten the test case, in my sandbox only, to replicate the recursive function using a differently spelled function name so i can spot it in the debug dump

i added these lines

+constexpr int Constexpr_recurse1(int n);
+
+constexpr int Constexpr_recurse(int n) {
+  if (n)
+    return Constexpr_recurse1(n - 1);
+  return 103;
+}
+
+constexpr int Constexpr_recurse1(int n) {
+  return Constexpr_recurse(n) + 1;
+}
+
Then I changed constexpr_recurse_test_err to call Constexpr_recurse

Reply:
silly me, i had the values swapped in the printf